app/src/main/java/unipd/se18/ocrcamera/PhotoTester.java

```diff
@@ -0,0 +1,449 @@
+
+            if (word.length() >= 3) { //ignoring non significant words with 1 or 2 characters
+                maxPoints += word.length();
+
+                for (int i = 0; i < extractedWords.length && !found; i++) {
+                    //for each extracted words starting from posLastWordFound
+                    index = (posLastWordFound + i) % extractedWords.length;
+
+                    //Calculate similarity and normalize
+                    int maxLength = Math.max(word.length(),extractedWords[index].length());
+                    double similarity = 1.0 - levenshtein.distance(word,extractedWords[index])/maxLength;
+
+                    //if similarity grater than 0.8 the word is found
+                    if (similarity > 0.8) {
+                        if (points == 0 || i < consecutiveNotFound + 10) {
+                            points += word.length()*similarity; //assign points based on number of characters
+                        } else {
+                            //if word found is distant from posLastWordFound the ocr text isn't ordered, less points
+                            points += (float) word.length()*similarity/2;
+                        }
+                        Log.d(TAG, "ingredientsTextComparison -> \"" + word + "\" ==  \"" + extractedWords[index] + "\" similarity="+similarity);
+                        extractedWords[index] = ""; //remove found word
+                        found = true;
+                    }
+                }
+            }
+
+            //taking into consideration words that are not properly separated (e.g. "cetarylalcohol")
+            if(!found && word.length() >= 6){
+                maxPoints += word.length();
+                for(int i=0; i<extractedWords.length && !found; i++) {
+                    index = (posLastWordFound+i)%extractedWords.length;
+                    if (extractedWords[index].contains(word)) { //the correct word is contained in the extracted word
+                        if(points==0 || i<consecutiveNotFound+10) {
+                            points += word.length(); //assign points based on number of characters
+                        } else {
+                            points += (float)word.length()/2;
+                        }
+                        Log.d(TAG, "ingredientsTextComparison -> \"" + word + "\" contained in  \"" + extractedWords[index] + "\"");
+                        extractedWords[index] = extractedWords[index].replace(word, ""); //remove found word
+                        found = true;
+                    }
+                }
+            }
+
+            if(found){
+                consecutiveNotFound = 0;
+                posLastWordFound = index;
+            } else {
+                consecutiveNotFound++;
+            }
+        }
+        float confidence = (points / maxPoints)*100;
+        Log.i(TAG, "ingredientsTextComparison -> confidence == " + confidence + " (%)");
+        return confidence;
+
+    }
```

Comment:
Excellent function.
I would advise you to use less magic numbers, because you use them very often and could create problems, and replace them with local variables.